code/processes/dqe.q

```diff
@@ -16,25 +16,80 @@ readdqeconfig:{[file]
 
 gethandles:{exec procname,proctype,w from .servers.SERVERS where (procname in x) | (proctype in x)};
 
-tableexists:{x in tables[]};                                                                                    /- function to check for table, param is table name as a symbol
+tableexists:{[tab]                                                                                              /- function to check for table, param is table name as a symbol
+  result:();
+  .lg.o[`dqe;"checking if ", (string tab), " table exists"];
+  $[1=a:tab in tables[];result:(1b;((string tab)," table exists"));result:(0b;(string tab)," missing from process")]
+  };
 
-runcheck:{[fn;vars;rs]                                                                                          /- function used to send other function to test processes
+tableticking:{[tab;timeperiod;timetype]                                                                         /- function to check if table is increasing in size
+  $[0<a:count select from tab where time within (.z.p-timetype$"J"$string timeperiod;.z.p);                     /- params are tab        - table to check
+    (1b;"there are ",(string a)," records");                                                                    /-            timeperiod - time back to check for records
+    (0b;"the table is not ticking")]                                                                            /-            timetype   - either `minute or `second
+  }
+
+chkslowsub:{[threshold]                                                                                         /- function to check for slow subscribers
+  .lg.o[`dqe;"Checking for slow subscribers"];
+  overlimit:(key .z.W) where threshold<sum each value .z.W;
+  $[0=count overlimit;
+    (1b;"no data queues over the limit, in ",(string .proc.procname));
+    (0b;raze"handle(s) ",("," sv string overlimit)," have queues")]
+  }
+
+fillprocname:{[h;rs]                                                                                            /- fill procname for results table
+  $[0=first where rs in ' h`procname`proctype;
+    enlist rs,'rs;
+    1=first where rs in ' h`procname`proctype;
+    rs,'exec procname from flip h where proctype=rs;enlist rs,'`]
```

Comment:
Need to update the indenting:
```
  $[0=first where rs in ' h`procname`proctype;
    enlist rs,'rs;
  1=first where rs in ' h`procname`proctype;
    rs,'exec procname from flip h where proctype=rs;
    enlist rs,'`]
```
But more importantly, what is this try to achieve? Looks ripe for simplification. Is this a symbol list?
```
h`procname`proctype
```

Reply:
h is a dictionary that is passed in and then two lists of symbols are extracted from it.
```
q.dqe))h
procname| rdb1 hdb1 hdb2
proctype| rdb  hdb  hdb
w       | 9    10   11
```
This function will create r
```
q.dqe))r
rdb rdb1
hdb hdb1
hdb hdb2
bbo
```
This will be used to fill the results table columns procs and chkprocs. Anywhere there is a process that was not able to be connected to, chkprocs will be a null.